src/server/components/lists/questionnaire.ts

```diff
@@ -271,4 +286,224 @@ export const questions: Questions = {
+        hintText = `${hintTextStart} translate into English.`;
+
+      } else if (servicesProvided?.includes("interpretation") && !servicesProvided?.includes("translation") && !servicesProvided?.includes("All") ) {
+        hintText = `${hintTextStart} interpret into English.`;
+      }
+      return hintText;
+    },
+    needsToAnswer(req: Request) {
+      const { languagesProvided, languagesPopulated } = getAllRequestParams(req);
+      return  !languagesPopulated || languagesProvided === undefined || languagesProvided === "";
+    },
+    getPartialData(req) {
+      const partialData: QuestionData[] = [];
+      Object.entries(languages).forEach(([key, value]) => {
+        partialData.push({
+          text: value,
+          value: key,
+        })
+      });
+      return partialData;
+    },
+    validate(req: Request) {
+      const { languagesProvided, languagesPopulated } = getAllRequestParams(req);
+
+      if (languagesProvided === "" && languagesPopulated) {
+        return {
+          field: "languagesProvided",
+          text: "You must select the language(s) you need translating or interpreting",
+          href: "#languages-provided-yes",
+        };
+      }
+      return false;
+    },
+  },
+  languagesSummary: {
+    getViewPartialName() {
+      return "questions/question-languages-summary.njk";
+    },
+    pageTitle() {
+      return "You have selected these languages";
+    },
+    needsToAnswer(req: Request) {
+      const { languagesProvided, languagesConfirmed } = getAllRequestParams(req);
+      return  !languagesConfirmed || languagesProvided === undefined || languagesProvided === "";
+    },
+    getPartialData(req) {
+      const partialData: QuestionData[] = [];
+      Object.entries(languages).forEach(([key, value]) => {
+        partialData.push({
+          text: value,
+          value: key,
+        })
+      });
+      return partialData;
+    },
+    validate(req: Request) {
+      const { languagesProvided, languagesConfirmed } = getAllRequestParams(req);
+
+      if (languagesProvided === "" && languagesConfirmed) {
+        return {
+          field: "languagesProvided",
+          text: "You must select the language(s) you need translating or interpreting",
+          href: "#languages-provided-yes",
+        };
+      }
+      return false;
+    },
+  },
+  translationSpecialties: {
+    getViewPartialName() {
+      return "questions/question-translation-specialties.njk";
+    },
+    pageTitle() {
+      return "What type of translation do you need?";
+    },
+    needsToAnswer(req: Request) {
+      const { servicesProvided, translationSpecialties, interpreterServices } = getAllRequestParams(req);
+      const result: boolean = (!translationSpecialties
+        && !!interpreterServices
+        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+        && (servicesProvided?.includes("translation") || servicesProvided?.includes("All"))
+      ) as boolean;
```

Comment:
I don't know if this `as boolean` type assertion is necessary if there is already a `boolean` type annotation on line **413**.

Was the type annotation not being picked up properly?

Reply:
minor but refactored to use nullish coalescing operator instead.